src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java

```diff
@@ -31,24 +31,27 @@
 
 public class UnmodifiableHeaders extends Headers {
 
-    private final Headers headers;  // mutable, but no reference to it escapes
-    private final Map<String, List<String>> map;
+    private final Headers headers;  // modifiable, but no reference to it escapes
+    private final Map<String, List<String>> map;  // unmodifiable
 
     public UnmodifiableHeaders(Headers headers) {
         var h = headers;
-        Map<String, List<String>> unmodListMap = new HashMap<>();
-        h.forEach((k, v) -> unmodListMap.put(k, Collections.unmodifiableList(v)));
-        this.map = Collections.unmodifiableMap(unmodListMap);
+        var unmodHeaders = new Headers();
+        h.forEach((k, v) -> {
+                List<String> l = v == null ? List.of() : Collections.unmodifiableList(v);
```

Comment:
If v is `null` it should be mapped to `null` - not to `List.of()`, to preserve map equality.